frappe/core/doctype/submission_queue/submission_queue.py

```diff
@@ -128,44 +132,21 @@ def notify(self, submission_status: str, action: str):
 			notify_to = frappe.db.get_value("User", self.enqueued_by, fieldname="email")
 			enqueue_create_notification([notify_to], notification_doc)
 
-	def _unlock_reference_doc(self):
-		"""
-		Only execute if self.job_id is defined.
-		"""
-		try:
-			job = Job.fetch(self.job_id, connection=get_redis_conn())
-			status = job.get_status(refresh=True)
-			exc = job.exc_info
-		except NoSuchJobError:
-			exc = None
-			status = "failed"
-
-		if status in ("queued", "started"):
-			frappe.msgprint(_("Document in queue for execution!"))
-			return
-
-		self.queued_doc.unlock()
-		values = (
-			{"status": "Finished"} if status == "finished" else {"status": "Failed", "exception": exc}
-		)
-		frappe.db.set_value(self.doctype, self.name, values, update_modified=False)
-		frappe.msgprint(_("Document Unlocked"))
-
```

Comment:
made `job_id` a link field to `RQ Job` doctype - we dont really need this much complexity anymore :P